drivers/soundwire/intel_init.c

```diff
@@ -219,9 +253,9 @@ static struct sdw_intel_ctx
 		return NULL;
 
 	ctx->count = count;
-	ctx->links = devm_kcalloc(&adev->dev, ctx->count,
-				  sizeof(*ctx->links), GFP_KERNEL);
-	if (!ctx->links)
+	ctx->ldev = devm_kcalloc(&adev->dev, ctx->count,
+				 sizeof(*ctx->ldev), GFP_KERNEL);
```

Comment:
@plbossart this still makes me a bit nervous about free after use errors. Basically devm_ would mean that the ldev would only be freed when the adev->dev is unregistered which i would assume be sometime after the ancillary device is unregistered. This would mean that as long as ldev is around, ldev->ancidev->dev can still be attempted to be accessed

Reply:
@ranj063 the ancillally device unregister is called from the PCI probe remove, and it's just the same as before when we used platform devices.